pkg/runtime/types.go

```diff
@@ -80,17 +86,19 @@ type TypeMeta struct {
 //
 // +protobuf=true
 type RawExtension struct {
-	// RawJSON is the underlying serialization of this object.
-	RawJSON []byte
+	// Raw is the underlying serialization of this object.
```

Comment:
We're trying to remove the conversion logic in pkg/runtime/embedded.go (conversions doesn't know enough about the context that a serialization is happening in to properly convert arbitrary objects to bytes).  In OpenShift we have more cases where we use embedded RawExtensions, so we hit the use cases more.  @deads2k and I ended up having an extension to the serializer that gave objects the chance to encode their nested objects with a copy of the serializer, prior to actually writing out the actual object.  In that case, the encoding and type would be handled by the serializer code.  

Example - we have a `List` object that has runtime.Objects.  Those objects don't have to be in the same API group, or be targeted to the same API version.  When we read the List, we don't decode them immediately (we just move them to unknown).  The serializer that reads those knows what content type is the default - conversion does not.  Likewise, when we write those to protobuf, the protobuf serializer knows that the destination content type is protobuf, not conversion.  So the hooks in the serializer have to go and give each object a chance to merge.

If we get rid of internal versions, we'd need a way to pass around objects that have been decoded (that's what Object is for).  I'm not sure that we'd need content encoding or type.

I think line 102 of embedded up above is technically wrong (where we convert unknown to raw extension), because we don't want conversion doing encoding.  The problem is we need to formalize how we do encoding of nested objects, probably as part of this chain of work.  Since Unknown is a wrapper, rather than an object on its own, we should probably